ddtrace/debugging/_uploader.py

```diff
@@ -56,6 +56,7 @@ def __init__(self, interval: Optional[float] = None) -> None:
         }
 
         container.update_headers_with_container_info(self._headers, container.get_container_info())
+        container.update_header_with_external_info(self._headers)
```

Comment:
Would it be possible to have this container logic encapsulated in https://github.com/p403n1x87/dd-trace-py/blob/dad4a9f079336d3bd0d35af3699228cc4ac840d2/ddtrace/internal/http.py#L33 instead of having to replicate it everywhere a connection object is used?

Reply:
It might be possible but I am not sure as I am not familiar with the codebase. The current implementation reflects what was done for `container.update_headers_with_container_info()`.

I would argue that if we want to encapsulate the container logic in `HTTPSConnection`, it should be done in another PR, WDYT?